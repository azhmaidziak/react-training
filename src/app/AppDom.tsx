import * as React from 'react';
import MenuDom from '../menu';
import BodyDom from "../body";
import {Page} from "../models";

export default class AppDom extends React.Component<Page> {
    constructor(page:Page){
        super(page);
        this.state = {
            count: 0
        }
    }

    render() {
        return <React.Fragment>
            <div className="menu">
                <MenuDom items={this.props.menu.items}/>
            </div>
            <div className="body">
                <BodyDom bodyText={this.props.body.bodyText}/>
            </div>
        </React.Fragment>;
    }
}