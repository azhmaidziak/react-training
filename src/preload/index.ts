import { Page } from "../models";

export const page: Page = {
  menu: { items: [{ key: "1", text: "Menu 1" }, { key: "2", text: "Menu 2" }] },
  body: { bodyText: "Hello React!" }
};

export const template = (html: string): string =>
  `<!DOCTYPE html>
    <html lang="en">
      <head>
        <meta charset="utf-8" />
        <meta name="theme-color" content="#000000" />
        <title>Movies Application</title>
      </head>      
      <body>
        <div id="root">${html}</div>
      <script src="/assets/client.js"></script>
      </body>    
    </html>`;
