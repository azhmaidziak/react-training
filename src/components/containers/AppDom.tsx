import * as React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import {page, search} from "../../preload";
import SearchContainer from "./SearchContainer";
import MoviesContainer from "./MoviesContainer";
import ErrorBoundary from "../renders/ErrorBoundary";
import configureStore from "../../store/configureStore";
import * as styles from "./App.module.sass"


export default class AppDom extends React.Component<{init:boolean}> {
  render(): React.ReactNode {
    return (
      <Provider store={configureStore(page, this.props.init)}>
        <div className={styles.app_body}>
          <ErrorBoundary>
            <SearchContainer />
            <MoviesContainer />
          </ErrorBoundary>
        </div>
      </Provider>
    );
  }
}
