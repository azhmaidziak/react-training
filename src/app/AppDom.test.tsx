import * as React from "react";
import AppDom from "../app";
import * as preload from "../preload";

import {
  create,
  ReactTestRenderer,
  ReactTestRendererJSON
} from "react-test-renderer";
import { cleanup } from "@testing-library/react";

describe("AppDom", () => {
  const { page } = preload;
  afterEach(cleanup);

  it("just test create component.", () => {
    const component: ReactTestRenderer = create(
      <AppDom menu={page.menu} body={page.body} />
    );
    const testInstance: ReactTestRendererJSON = component.toJSON();
    expect(testInstance).toMatchSnapshot();
  });
});
