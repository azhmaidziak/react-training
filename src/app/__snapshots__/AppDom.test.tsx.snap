// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDom just test create component. 1`] = `
Array [
  <div
    className="app_menu"
  >
    <ul>
      <li
        onClick={[Function]}
      >
        Menu 1
        : with count = [
        0
        ]
      </li>
      <li
        onClick={[Function]}
      >
        Menu 2
        : with count = [
        0
        ]
      </li>
    </ul>
  </div>,
  <div
    className="app_body"
  >
    <div
      className="dark"
    >
      <p
        className="content"
      >
        Hello React!
      </p>
    </div>
  </div>,
]
`;
